src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py

```diff
@@ -334,23 +334,23 @@ def set_key(cmd,
             raise CLIError("Failed to retrieve key-values from config store. " + str(exception))
 
         if retrieved_kv is None:
-            if content_type and __is_json_content_type(content_type):
+            if __is_json_content_type(content_type):
                 try:
                     # Ensure that provided value is valid JSON. Error out if value is invalid JSON.
-                    value = 'null' if value is None else value
+                    value = '""' if value is None else value
```

Comment:
Since this is a JSON key-value, the default value should be null because we don't want to assume that its a string value.

Reply:
Reverted.